pkg/gds/client/profiles.go

```diff
@@ -75,69 +154,51 @@ func LoadActive(c *cli.Context) (p *Profile, err error) {
 	return p, nil
 }
 
-// Active returns the profile with the specified name or the active profile if no name
-// is specified. If multiple profiles are marked active it returns the first active
-// profile with no ordering guarantees. If no profiles are marked active and there is
-// one profile, that profile is returned, otherwise an error is returned.
-func (p Profiles) Active(name string) (_ *Profile, err error) {
+// GetActive returns the profile with the specified name or the active profile if no name
+// is specified.
+func (p Profiles) GetActive(name string) (_ *Profile, err error) {
 	if name != "" {
-		profile, ok := p[name]
+		profile, ok := p.Profiles[name]
 		if !ok {
 			return nil, fmt.Errorf("no profile found named %q", name)
 		}
 		return profile, nil
 	}
 
-	for _, profile := range p {
-		if profile.Active {
-			return profile, nil
-		}
-	}
-
-	if len(p) == 1 {
-		for _, profile := range p {
-			return profile, nil
-		}
-	}
-
-	return nil, fmt.Errorf("no active profile found in %d profiles", len(p))
+	return p.Profiles[p.Active], nil
 }
 
 // SetActive marks the profile with the specified name as active.
 func (p Profiles) SetActive(name string) (err error) {
-	if _, ok := p[name]; !ok {
+	if _, ok := p.Profiles[name]; !ok {
 		return fmt.Errorf("no profile named %q found", name)
 	}
 
-	// Mark all profiles inactive to ensure only one profile is active at a time
-	for _, profile := range p {
-		profile.Active = false
-	}
-
-	// Mark the specified profile as active
-	p[name].Active = true
+	p.Active = name
 
 	// Save the profiles back to disk to ensure the activation takes effect
-	if err = p.Save(); err != nil {
+	if err = p.Save(nil); err != nil {
 		return fmt.Errorf("could not save active profile: %s", err)
 	}
 	return nil
 }
 
-// Save the profiles to disk in the configuration directory.
-func (p Profiles) Save() (err error) {
-	folders := cfgd.QueryFolders(configdir.Global)
-	if len(folders) == 0 {
-		return errors.New("could not find user configuration directory")
+// Save the profiles to disk in the specified configuration folder. If the configuration
+// folder is nil, the configuration folder is located and created if it doesn't exist.
+func (p Profiles) Save(folder *configdir.Config) (err error) {
+	if folder == nil {
+		if folder, err = GetProfilesFolder(); err != nil {
+			return fmt.Errorf("could not find profiles folder: %s", err)
+		}
```

Comment:
Because the search order starts with the local directory, will this create the config folder in the CWD? I think the default should be in the user folder, but let me know if I'm missing something here.

Reply:
You're right, that's the current behavior. I think defaulting to the user folder does make more sense, so maybe I'll make the write precedence user -> global -> local and then use the library's local -> user -> global search order when looking for the config file so that the user can still create an overriding config in the CWD.